### DISCUSSION TITLE
ButtonsMessage Alternatives

### DISCUSSION THREAD

**Discussion by mamur-rezeki** [tulir/whatsmeow#237] [WhatsApp protocol Q&A]
Since the ButtonsMessage become very strange, I've been testing alternative way to send ButtonsMessage:

1. Wrap with ` ViewOnceMessage ` 

 ```go
var viewOnce = &waProto.Message{
  ViewOnceMessage: &waProto.FutureProofMessage{
    Message: msgbutton,
    },
}

```

**Comment by Diden05** [comment]
Yes, it works, tested for a month

**Comment by bhavesh53** [comment]
not working for iPhone . working for Android .

**Reply by Diden05** [reply]
> not working for iPhone . working for Android .

Not working only template message, button and list message working

**Reply by ayrokid** [reply]
how about a template message always on the left position when sending a message?

**Reply by Diden05** [reply]
It is, and I don't know how to fix it.

**Reply by ghost** [reply]
Any solution?
